packages/edit-site/src/hooks/commands/use-common-commands.js

```diff
@@ -237,21 +238,32 @@ function useGlobalStylesOpenCssCommands() {
 }
 
 function useGlobalStylesOpenRevisionsCommands() {
-	const { openGeneralSidebar, setEditorCanvasContainerView, setCanvasMode } =
-		unlock( useDispatch( editSiteStore ) );
-	const { getCanvasMode } = unlock( useSelect( editSiteStore ) );
+	const {
+		openGeneralSidebar,
+		closeGeneralSidebar,
+		setEditorCanvasContainerView,
+		setCanvasMode,
+	} = unlock( useDispatch( editSiteStore ) );
+	const { getCanvasMode, getEditorCanvasContainerView } = unlock(
+		useSelect( editSiteStore )
+	);
```

Comment:
My understanding is that instead of unlocking `useSelect` itself, you will need to do it _inside_ the `useSelect` hook. You will probably need to modify the existing _useSelect_ hook to look like this:

```javascript
	const { canvasMode, hasRevisions, isStyleRevisionsOpened } = useSelect(
		( select ) => {
			const { getEntityRecord, __experimentalGetCurrentGlobalStylesId } =
				select( coreStore );
			const { getCanvasMode, getEditorCanvasContainerView } = unlock(
				select( editSiteStore )
			);
			const globalStylesId = __experimentalGetCurrentGlobalStylesId();
			const globalStyles = globalStylesId
				? getEntityRecord( 'root', 'globalStyles', globalStylesId )
				: undefined;
			return {
				canvasMode: getCanvasMode(),
				hasRevisions:
					!! globalStyles?._links?.[ 'version-history' ]?.[ 0 ]
						?.count,
				isStyleRevisionsOpened:
					'global-styles-revisions' ===
					getEditorCanvasContainerView(),
			};
		},
		[]
	);
```

Reply:
I agree with the `getEditorCanvasContainerView` move inside the `useSelect`, as it's being used there.

`getCanvasMode` though is used in a callback function, so it's fine to have outside and call it when needed to get the latest value.

Reply:
If you have a few moments, feel free to take over the PR to get this in.